waspc/data/Generator/templates/react-app/src/queries/core.d.ts

```diff
@@ -1,11 +1,21 @@
 import { type Query } from '.'
+import { Route } from '../types';
 import type { Expand, _Awaited } from '../universal/types'
 
 export function createQuery<BackendQuery extends GenericBackendQuery>(
   queryRoute: string,
   entitiesUsed: any[]
 ): QueryFor<BackendQuery>
 
+export function addMetadataToQuery(
+  query: (...args: any[]) => Promise<unknown>,
```

Comment:
@sodic The only thing I'm unsure about, please check if I've typed it as "any async function"

Reply:
This function receives _some function_ that it attaches additional props to. It was usually defined inside of `makeQuery` and it isn't really connected with `Query`